### PR TITLE
#2440 filter scores attached to deleted judges from scores datagrid

### PR DESCRIPTION
**Pull request by stenington**
This one wasn't as bad as I thought it might be. Initially I was just adding a check for `judge_profiles.deleted_at IS NULL` but that wasn't guaranteeing the presence of a judge profile because the sql query is using a LEFT OUTER JOIN, so in some cases **all** the judge profile fields were null. Asserting that we do have an id and we don't have the deleted flag works though.

### CODE REVIEW

File: app/models/submission_score.rb [hunk 1]
```diff
@@ -182,6 +182,12 @@ class SubmissionScore < ActiveRecord::Base
     end
   }
 
+  scope :judge_not_deleted, -> {
+    includes(:judge_profile)
+      .references(:judge_profiles)
+      .where("judge_profiles.id IS NOT NULL AND judge_profiles.deleted_at IS NULL")
```
Comment (shaunxp20) [inline app/models/submission_score.rb:188]: Huh, that `judge_profiles.id IS NOT NULL` part is strange! 🤔 😕 Fwiw, I tried a left join with just `judge_profiles.deleted_at IS NULL` in a manual/non-Rails SQL query locally and it works as expected. I'm going to guess that `judge_profiles.id IS NOT NULL` is necessary b/c of the paranoia soft-delete gem is interfering w/ the query. 🤔 ❔  🤷 Anywho, looks good to me, and it works! And tests! 👍 💯 :shipit:

Comment (stenington) [reply app/models/submission_score.rb:188]: My understanding is that references + includes does a LEFT OUTER JOIN, which is why you have to affirm you have an id.

Comment (shaunxp20) [reply app/models/submission_score.rb:188]: Looks like Paranoia is the culprit, it's adding a clause to the join:

`LEFT OUTER JOIN "judge_profiles" ON "judge_profiles"."id" = "submission_scores"."judge_profile_id" AND "judge_profiles"."deleted_at" IS NULL`

When I was testing locally I just did a normal/non-paranoia join: `left join judge_profiles on submission_scores.judge_profile_id = judge_profiles.id`, (with `judge_profiles.deleted_at IS NULL` in the where clause too) and it works as expected w/o the `judge_profiles.id IS NOT NULL`. Anywho, I think it's fine. 👍

Comment (stenington) [reply app/models/submission_score.rb:188]: Ah, ok, makes sense. Thanks!

**Pull request merged**